utils/ESR_Peers_List.cpp

```diff
@@ -0,0 +1,317 @@
+
+
+    void
+    trim_right_nulls(std::string& s)
+    {
+        boost::algorithm::trim_right_if(s, [](auto &c){return c=='\0';});
+    }
+
+
+    bzn::json_message
+    str_to_json(const std::string &json_str)
+    {
+        bzn::json_message json_msg;
+        Json::CharReaderBuilder builder;
+        Json::CharReader* reader = builder.newCharReader();
+        std::string errors;
+        if(!reader->parse(
+                json_str.c_str()
+                , json_str.c_str() + json_str.size()
+                , &json_msg
+                , &errors))
+        {
+            throw (std::runtime_error(ERR_UNABLE_TO_PARSE_JSON_RESPONSE + errors));
+        }
+        return json_msg;
+    }
+
+
+    bzn::json_message
+    make_params(const std::string& to_hex, const std::string& data_hex)
+    {
+        bzn::json_message params;
+        bzn::json_message param;
+        param["to"] = "0x" + to_hex; //407cA38Ded7929843BCF1f00A3303FC7dFdc12fC";
+        param["data"] = /*"0x" +*/ data_hex; // 0x46e76d8b0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000d426c757a656c6c65537761726d00000000000000000000000000000000000000";
+        params.append(param);
+        params.append("latest");
+        return params;
+    }
+
+
+    std::string
+    make_request(const std::string& to_hex, const std::string& data_hex)
+    {
+        bzn::json_message request;
+        request["jsonrpc"] = "2.0";
+        request["method"] = "eth_call";
+        request["params"] = make_params(to_hex, data_hex);
+        request["id"] = 1;
+        Json::StreamWriterBuilder wbuilder;
+        wbuilder["indentation"] = "";
+        return Json::writeString(wbuilder, request);
+    }
+
+
+    std::string
+    hex_to_utf8(const std::string& hex)
+    {
+        std::stringstream utft8stream;
+        boost::algorithm::unhex(hex, std::ostream_iterator<char>{utft8stream, ""});
+        return utft8stream.str();
+    }
+
+
+    std::string
+    string_to_hex(const std::string& value)
+    {
+        std::stringstream hexstream;
+        boost::algorithm::hex(value, std::ostream_iterator<char>{hexstream, ""});
+        return hexstream.str();
+    }
+
+
+    std::vector<std::string>
+    get_peers_result_to_vector(const std::string_view &result)
+    {
+        std::vector<std::string> results;
+        std::vector<std::string> lines;
+        size_t node_count{0};
+        enum states {HEADER, HEADER_SWARM_SIZE, HEADER_INFO, PEER_ID_SIZE, PEER_ID } state{HEADER};
+
+        char line[65]{0};
```

Comment:
65?

Reply:
Right, 64 will do. I should also not use the magic number...

Reply:
64 did not work as the unhex function choked on a non 0 terminated string... Changed the magic numbers to a const.